### PR TITLE
Net223 rider-79869 IBaker Usage inscpection suppression

### PR DESCRIPTION
**Pull request by jorgeamado**
https://youtrack.jetbrains.com/issue/RIDER-84159/Class-derived-from-Baker-should-be-marked-as-used

```
using Unity.Entities;
using Unity.Rendering;

class CannonBallAuthoring : UnityEngine.MonoBehaviour
{
}

class CannonBallBaker : Baker<CannonBallAuthoring>
{
    public override void Bake(CannonBallAuthoring authoring)
    {
        // By default, components are zero-initialized.
        // So in this case, the Speed field in CannonBall will be float3.zero.
        AddComponent<CannonBall>();
        AddComponent<URPMaterialPropertyBaseColor>();
    }
}
```

### CODE REVIEW

**Comment by jorgeamado** [comment]
Please review after https://github.com/JetBrains/resharper-unity/pull/2369

File: resharper/resharper-unity/src/Unity/CSharp/Daemon/UsageChecking/UsageInspectionsSuppressor.cs [hunk 1]
```diff
@@ -59,7 +59,10 @@ public bool SuppressUsageInspectionsOnElement(IDeclaredElement element, out Impl
                                      IsUxmlFactory(cls):
                     flags = ImplicitUseKindFlags.InstantiatedNoFixedConstructorSignature;
                     return true;
-
+                case IStruct @struct when unityApi.IsUnityType(@struct) ||
```
Comment (krasnotsvetov) [inline resharper/resharper-unity/src/Unity/CSharp/Daemon/UsageChecking/UsageInspectionsSuppressor.cs:62]: 1. Why only struct? System also could be class
2. Why are you setting implicit use for any "IsUnityType"?

File: resharper/resharper-unity/src/Unity/UnityEditorIntegration/Api/UnityApi.cs [hunk 2]
```diff
@@ -55,7 +55,9 @@ public UnityApi(UnityVersion unityVersion, UnityTypeCache unityTypeCache, UnityT
             type != null && myUnityTypeCache.IsUnityType(type);
 
         public static bool IsDotsSystemType([NotNullWhen(true)] ITypeElement? typeElement) =>
```
Comment (krasnotsvetov) [inline resharper/resharper-unity/src/Unity/UnityEditorIntegration/Api/UnityApi.cs:57]: IsDotsSystemType -> IsDotsType (baker is not system)

**Review by krasnotsvetov — CHANGES_REQUESTED** [review]

File: resharper/resharper-unity/src/Unity/CSharp/Daemon/UsageChecking/UsageInspectionsSuppressor.cs (hunk 1 above)
Comment (jorgeamado) [reply resharper/resharper-unity/src/Unity/CSharp/Daemon/UsageChecking/UsageInspectionsSuppressor.cs:62]: Class is processed on the line 55

**Comment by jorgeamado** [comment]
Will be merged via TeamCity pipeline

**Pull request closed without merging**